distributed/tests/test_worker.py

```diff
@@ -444,32 +444,79 @@ async def test_io_loop(s):
 @gen_cluster(client=True, nthreads=[])
 async def test_spill_to_disk(c, s):
     np = pytest.importorskip("numpy")
-    w = await Worker(
+    async with Worker(
         s.address,
         loop=s.loop,
         memory_limit=1200 / 0.6,
         memory_pause_fraction=None,
         memory_spill_fraction=None,
-    )
+    ) as w:
 
-    x = c.submit(np.random.randint, 0, 255, size=500, dtype="u1", key="x")
-    await wait(x)
-    y = c.submit(np.random.randint, 0, 255, size=500, dtype="u1", key="y")
-    await wait(y)
+        x = c.submit(np.random.randint, 0, 255, size=500, dtype="u1", key="x")
```

Comment:
```suggestion
        x = c.submit(np.zeros, 500, dtype="u1", key="x")
```
+ everywhere else below